core/jobs/batch_jobs/manual_voice_artist_name_job_test.py

```diff
@@ -0,0 +1,836 @@
+from core.domain import exp_services
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.jobs import job_test_utils
+from core.jobs.batch_jobs import manual_voice_artist_name_job
+from core.jobs.types import job_run_result
+from core.platform import models
+from core.tests import test_utils
+
+from typing import Dict, Type
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import voiceover_models
+
+(voiceover_models, exp_models) = models.Registry.import_models([
+    models.Names.VOICEOVER, models.Names.EXPLORATION])
+
+
+class VoiceArtistMetadataModelsTestsBaseClass(
+    job_test_utils.JobTestBase, test_utils.GenericTestBase):
+
+    EDITOR_EMAIL_1 = 'editor1@example.com'
+    EDITOR_EMAIL_2 = 'editor2@example.com'
+    EDITOR_EMAIL_3 = 'editor3@example.com'
+    EDITOR_EMAIL_4 = 'editor4@example.com'
+    EDITOR_EMAIL_5 = 'editor5@example.com'
+    EDITOR_USERNAME_1 = 'editor1'
+    EDITOR_USERNAME_2 = 'editor2'
+    EDITOR_USERNAME_3 = 'editor3'
+    EDITOR_USERNAME_4 = 'editor4'
+    EDITOR_USERNAME_5 = 'editor5'
+
+    CURATED_EXPLORATION_ID_1 = 'exploration_id_1'
+    CURATED_EXPLORATION_ID_2 = 'exploration_id_2'
+    NON_CURATED_EXPLORATION_ID_3 = 'exploration_id_3'
+    CURATED_EXPLORATION_ID_4 = 'exploration_id_4'
+
+    TOPIC_ID_1 = 'topic_id_1'
+    TOPIC_ID_2 = 'topic_id_2'
+    STORY_ID_1 = 'story_id_1'
+    STORY_ID_2 = 'story_id_2'
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.EDITOR_EMAIL_1, self.EDITOR_USERNAME_1)
+        self.signup(self.EDITOR_EMAIL_2, self.EDITOR_USERNAME_2)
+        self.signup(self.EDITOR_EMAIL_3, self.EDITOR_USERNAME_3)
+        self.signup(self.EDITOR_EMAIL_4, self.EDITOR_USERNAME_4)
+        self.signup(self.EDITOR_EMAIL_5, self.EDITOR_USERNAME_5)
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+
+        self.set_curriculum_admins([
+            self.EDITOR_USERNAME_1,
+            self.EDITOR_USERNAME_2,
+            self.EDITOR_USERNAME_3,
+            self.EDITOR_USERNAME_4,
+            self.EDITOR_USERNAME_5,
+            self.CURRICULUM_ADMIN_USERNAME])
+
+        self.editor_id_1 = self.get_user_id_from_email(self.EDITOR_EMAIL_1)
+        self.editor_id_2 = self.get_user_id_from_email(self.EDITOR_EMAIL_2)
+        self.editor_id_3 = self.get_user_id_from_email(self.EDITOR_EMAIL_3)
+        self.editor_id_4 = self.get_user_id_from_email(self.EDITOR_EMAIL_4)
+        self.editor_id_5 = self.get_user_id_from_email(self.EDITOR_EMAIL_5)
+        self.admin_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+    def _create_curated_explorations(self) -> None:
+        """The method generates two curated explorations and integrates them
```

Comment:
There seem to be more than two curated explorations being created in this method.